test/e2e/session-token.spec.js

```diff
@@ -0,0 +1,147 @@
+    const cliPath = path.join(__dirname, '../../src/bin/index.js');
+    exec(`node ${cliPath} ${command}`, { cwd: projectPath }, (error, stdout, stderr) => {
+      if (error) {
+        console.error(stderr);
+        reject(stdout);
+      } else {
+        resolve(stdout);
+      }
+    });
+  });
+};
+
+describe('e2e/session-token', function() {
+  this.timeout(15000);
+  
+  let sessionToken;
+  let action = 'upload-docs --force';
+
+  const initializeProject = async () => {
+    await runCliCommand('initialise-project-layout');
+  };
+
+  const initializeDatabase = async () => {
+    PouchDB.plugin(require('pouchdb-adapter-http'));
+    PouchDB.plugin(require('pouchdb-mapreduce'));
+    const pouchDb = new PouchDB(`${COUCHDB_URL}/medic`);
+
+    await pouchDb.put({
+      _id: '_design/medic',
+      views: {}
+    });
+  };
+
+  const createAdminUser = async () => {
+    const pouchDb = new PouchDB(`${COUCHDB_URL}/_users`);
+    const userDoc = {
+      _id: `org.couchdb.user:${COUCHDB_USERNAME}`,
+      name: COUCHDB_USERNAME,
+      roles: ['_admin'],
+      type: 'user',
+      password: COUCHDB_PASSWORD
+    };
+    await pouchDb.put(userDoc);
+  };
+
+  const getSessionToken = async () => {
+    const options = {
+      method: 'POST',
+      uri: 'http://localhost:6984/_session',
+      body: {
+        name: COUCHDB_USERNAME,
+        password: COUCHDB_PASSWORD,
+      },
+      resolveWithFullResponse: true,
+      json: true
+    };
+  
+    try {
+      const response = await rpn(options);
+      const setCookieHeader = response.headers['set-cookie'];
+      // Extract the session token from the set-cookie header
+      const sessionCookie = setCookieHeader.find(cookie => cookie.startsWith('AuthSession='));
+      const sessionToken = sessionCookie.split(';')[0].split('=')[1];
+      return sessionToken;
+    } catch (error) {
+      throw new Error(`Failed to get session token: ${error.message}`);
+    }
+  };
+
+  before(async () => {
+    createProjectPath();
+    await initializeDatabase();
+    await initializeProject();
+    await createAdminUser();
+    sessionToken = await getSessionToken();
+  });
+
+  after(cleanProjectPath);
+
+  it('should handle authentication with session token', async () => {
+    const stdout = await runCliCommand(
+      `--url=${COUCHDB_URL} --session-token=${sessionToken} ${action}`
+    );
+    expect(stdout).to.contain('INFO upload-docs complete.');
+  });
+
+  it('should handle authentication with basic authentication', async () => {
+    const url = `http://${COUCHDB_USERNAME}:${COUCHDB_PASSWORD}@localhost:6984`;
+    const stdout = await runCliCommand(
+      `--url=${url} ${action}`
+    );
+    expect(stdout).to.contain('INFO upload-docs complete.');
+  });
+
+  it('should fail with incorrect session token', async () => {
+    const incorrectToken = 'incorrect-token';
+    try {
+      await runCliCommand(`--url=${COUCHDB_URL} --session-token=${incorrectToken} ${action}`);
```

Comment:
In order for this test to be correct, you need to add a failed assertion here:
```
expect.fail('should have thrown');
```

Otherwise, if your command succeeds, you'll never enter the catch block and the test will just pass, instead of failing. 
Another option is to use chai-as-promised and not use try-catch blocks at all.